// @nrwl/vite/src/generators/init/init.js
"use strict";
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    createVitestConfig: ()=>createVitestConfig,
    initGenerator: ()=>initGenerator,
    default: ()=>_default,
    initSchematic: ()=>initSchematic
});
const _devkit = require("@nrwl/devkit");
const _versions = require("../../utils/versions");
var _workspaceConfiguration, _targetDefaults, _test;
function checkDependenciesInstalled(host, schema) {
    const packageJson = (0, _devkit.readJson)(host, 'package.json');
    const devDependencies = {};
    const dependencies = {};
    packageJson.dependencies = packageJson.dependencies || {};
    packageJson.devDependencies = packageJson.devDependencies || {};
    // base deps
    devDependencies['@nrwl/vite'] = _versions.nxVersion;
    devDependencies['vite'] = _versions.viteVersion;
    devDependencies['vite-plugin-eslint'] = _versions.vitePluginEslintVersion;
    devDependencies['vite-tsconfig-paths'] = _versions.viteTsConfigPathsVersion;
    devDependencies['vitest'] = _versions.vitestVersion;
    devDependencies['@vitest/ui'] = _versions.vitestUiVersion;
    devDependencies['jsdom'] = _versions.jsdomVersion;
    if (schema.uiFramework === 'react') {
        devDependencies['@vitejs/plugin-react'] = _versions.vitePluginReactVersion;
    }
    if (schema.includeLib) {
        devDependencies['vite-plugin-dts'] = _versions.vitePluginDtsVersion;
    }
    return (0, _devkit.addDependenciesToPackageJson)(host, dependencies, devDependencies);
}
function moveToDevDependencies(tree) {
    (0, _devkit.updateJson)(tree, 'package.json', (packageJson)=>{
        packageJson.dependencies = packageJson.dependencies || {};
        packageJson.devDependencies = packageJson.devDependencies || {};
        if (packageJson.dependencies['@nrwl/vite']) {
            packageJson.devDependencies['@nrwl/vite'] = packageJson.dependencies['@nrwl/vite'];
            delete packageJson.dependencies['@nrwl/vite'];
        }
        return packageJson;
    });
}
function createVitestConfig(tree) {
    var ref;
    const workspaceConfiguration = (0, _devkit.readWorkspaceConfiguration)(tree);
    const productionFileSet = (ref = workspaceConfiguration.namedInputs) == null ? void 0 : ref.production;
    if (productionFileSet) {
        productionFileSet.push('!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)', '!{projectRoot}/tsconfig.spec.json');
        workspaceConfiguration.namedInputs.production = Array.from(new Set(productionFileSet));
    }
    var _targetDefaults1;
    (_targetDefaults1 = (_workspaceConfiguration = workspaceConfiguration).targetDefaults) != null ? _targetDefaults1 : _workspaceConfiguration.targetDefaults = {};
    var _test1;
    (_test1 = (_targetDefaults = workspaceConfiguration.targetDefaults).test) != null ? _test1 : _targetDefaults.test = {};
    var _inputs;
    (_inputs = (_test = workspaceConfiguration.targetDefaults.test).inputs) != null ? _inputs : _test.inputs = [
        'default',
        productionFileSet ? '^production' : '^default'
    ];
    (0, _devkit.updateWorkspaceConfiguration)(tree, workspaceConfiguration);
}
function initGenerator(tree, schema) {
    moveToDevDependencies(tree);
    createVitestConfig(tree);
    const installTask = checkDependenciesInstalled(tree, schema);
    return installTask;
}
const _default = initGenerator;
const initSchematic = (0, _devkit.convertNxGenerator)(initGenerator);
